apis/auth-api/src/me/contract.controller.ts

```diff
@@ -26,8 +34,67 @@ export class ContractController {
 		description: "EOA address (0x...) to deploy smart contract as signer",
 		example: "0x19e7e376e7c213b7e7e46cc70a5dd086daff2a",
 	})
-	@ApiOkResponse({ type: ContractDeployResponse })
+	@ApiAcceptedResponse({
+		schema: {
+			type: "object",
+			properties: {
+				jobId: { type: "string" },
+				statusUrl: { type: "string" },
+			},
+		},
+	})
+	@Post(":owner")
+	@HttpCode(202)
 	async deploy(@Param("owner") owner: Address) {
-		return this.contractService.deployContractForOwner(owner);
+		const chainId = Number(process.env.CHAIN_ID);
+		const jobId = `deploy_${chainId}_${owner.toLowerCase()}`;
+
+		const existing = await this.deployQueue.getJob(jobId);
+		if (existing) {
+			return {
+				jobId,
+				statusUrl: `/contract/jobs/${encodeURIComponent(jobId)}`,
```

Comment:
curious why this is being URI encoded? none of the characters in the jobId would be escaped.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#description